ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/LedgerFactory.scala

```diff
@@ -148,21 +162,36 @@ trait LedgerFactory[+RWS <: ReadWriteService, ExtraConfig]
       config: Config[ExtraConfig],
       participantConfig: ParticipantConfig,
       engine: Engine,
-  )(implicit materializer: Materializer, loggingContext: LoggingContext): ResourceOwner[RWS] =
-    readWriteServiceOwner(config, participantConfig, engine)
+      completionService: IndexCompletionsService,
```

Comment:
So, this parameter (and the engine) are not actually used by the read service, only the write service. We therefore don't even need to construct these objects in "indexer" mode.

I think this is starting to tell us that it's no longer sensible to couple the reader and writer.

Reply:
Yes. As a matter of fact, I've initially started refactoring this but realized that it's not really that straightforward. 
The problem here is that the design is inherently wrong, and the composition flow is actually reversed, instead of creating a read-write service from the individual read service and write service, we create the individual read service and write service from the always composed read-write service. 
I will try to tackle this in a new PR

Reply:
Totally agree that it's out of scope here.